packages/api/cms-api/src/content-generation/content-generation.module.ts

```diff
@@ -1,29 +1,41 @@
-import { DynamicModule, Global, Module } from "@nestjs/common";
+import { DynamicModule, Global, Module, Type } from "@nestjs/common";
 
-import { CONTENT_GENERATION_SERVICE } from "./content-generation.constants";
-import { ContentGenerationServiceInterface } from "./content-generation.service.interface";
+import { CONTENT_GENERATION_CONFIG, CONTENT_GENERATION_SERVICE } from "./content-generation.constants";
+import { ContentGenerationServiceInterface } from "./content-generation-service.interface";
 import { GenerateAltTextResolver } from "./generate-alt-text.resolver";
 import { GenerateImageTitleResolver } from "./generate-image-title.resolver";
+import { OpenAiContentGenerationService } from "./openai-content-generation.service";
+
+export interface ContentGenerationModuleOptions<T> {
+    Service: Type<ContentGenerationServiceInterface>;
+    config: T;
```

Comment:
The ContentGenerationModule doesn't need to know about the config IMO. Also, it shouldn't provide the config

Reply:
Either call an `init` method or pass it as arguments to the separate methods, see https://github.com/vivid-planet/comet/pull/1896#discussion_r1543063148